packages/angular/cli/src/analytics/analytics.ts

```diff
@@ -246,57 +177,18 @@ export async function getGlobalAnalytics(): Promise<AnalyticsCollector | undefin
     if (process.env['NG_CLI_ANALYTICS'] === 'ci') {
       analyticsDebug('Running in CI mode');
 
-      return new AnalyticsCollector(propertyId, 'ci');
+      return new AnalyticsCollector(AnalyticsProperties.AngularCliDefault, 'ci');
     }
   }
 
-  // If anything happens we just keep the NOOP analytics.
-  try {
-    const globalWorkspace = await getWorkspace('global');
-    const analyticsConfig: string | undefined | null | { uid?: string } =
-      globalWorkspace && globalWorkspace.getCli() && globalWorkspace.getCli()['analytics'];
-    analyticsDebug('Client Analytics config found: %j', analyticsConfig);
-
-    if (analyticsConfig === false) {
-      analyticsDebug('Analytics disabled. Ignoring all analytics.');
-
-      return undefined;
-    } else if (analyticsConfig === undefined || analyticsConfig === null) {
-      analyticsDebug('Analytics settings not found. Ignoring all analytics.');
-
-      // globalWorkspace can be null if there is no file. analyticsConfig would be null in this
-      // case. Since there is no file, the user hasn't answered and the expected return value is
-      // undefined.
-      return undefined;
-    } else {
-      let uid: string | undefined = undefined;
-      if (typeof analyticsConfig == 'string') {
-        uid = analyticsConfig;
-      } else if (typeof analyticsConfig == 'object' && typeof analyticsConfig['uid'] == 'string') {
-        uid = analyticsConfig['uid'];
-      }
-
-      analyticsDebug('client id: %j', uid);
-      if (uid == undefined) {
-        return undefined;
-      }
-
-      return new AnalyticsCollector(propertyId, uid);
-    }
-  } catch (err) {
-    analyticsDebug('Error happened during reading of analytics config: %s', err.message);
-
-    return undefined;
-  }
+  return getAnalytics('global');
 }
 
-export async function hasWorkspaceAnalyticsConfiguration(): Promise<boolean> {
+export async function hasAnalyticsConfiguration(level: 'global' | 'local'): Promise<boolean> {
```

Comment:
Nit: Could we simplify this implementation to `return !!(await getWorkspaceAnalytics());`? Could we replace calls to `hasAnalyticsConfiguration()` to just `!!(await getWorkspaceAnalytics());` and drop the function altogether? That might make some of the callers' logic simpler if they just try to get the analytics object and check if they actually got one.

Reply:
Yes, that can be done.

Reply:
Actually checking the logic more, we cannot use `getWorkspaceAnalytics` instead of `hasAnalyticsConfiguration` has we need to differentiate between being disabled and unset in which `getWorkspaceAnalytics` will always return undefined in this case.

Reply:
One option might be to have `getWorkspaceAnalytics()` return a discriminated union or return `Analytics | 'disabled' | 'unset'`.

It's up to you, I'm fine with `hasWorkspaceAnalyticsConfiguration()` if that's the best approach, just suggesting an alternative to consider.